one-ui/src/main/java/com/marklogic/hub/oneui/utils/DataHubProjectUtils.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2012-2020 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.marklogic.hub.oneui.utils;
+
+import com.marklogic.hub.HubProject;
+import com.marklogic.hub.deploy.util.HubDeployStatusListener;
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.multipart.MultipartFile;
+
+public class DataHubProjectUtils {
```

Comment:
I'd much rather see a real class here as opposed to a collection of static methods. This looks more like a "ProjectManager" class with methods for backing up a project, cleaning a project, and extracting a project zip. 

That also makes it easy to write tests for these methods so that we know they work.

Reply:
I think that those methods can be used as utilty methods and do not feel we should create instances for that. I did not see any difference from creating "ProjectManager" class. We also can create test cases, right?

Reply:
First, I'll take back my suggestion, as since this is bound to HubDeployStatusListener, there may not be much reuse to achieve here. 

Second - generally, I only like static functions/variables for things that are truly "static" - i.e. not likely to change. Statics also create tight binding between classes - i.e. there's no way to modify a client's usage of a static. For example, if we instead had a ProjectManager interface, the EnvironmentController (or DataHubService) could be bound to that interface. And it could have a constructor or setter that allows for any impl of that interface to be passed in. That results in highly composable, reusable software. Statics prevent that.